### PR TITLE
Fix : code duplication and item condition reset

### PR DESCRIPTION
**Pull request by MarcK0909**
## Summary
This PR aims at having a shared item-form UI logic, exposing a “clear” option for condition selection, and adding JVM/UI tests to cover the refactor plus the new mapping utilities. I also reduced the code duplication which was an issue in my last PR #358. 

### UI
- `AddItemsScreen`, `EditItemsScreen`, and `ItemFieldsComponents` now use the new form scaffolding, provide shared additional-details tags, and expose a **“Clear condition”** action in the dropdown.
- Added `ItemFormSharedComponents` to host the reusable accordion/layout code powering both item screens.

### Tests
- Added **Robolectric Compose tests** (`ItemFormSharedComponentsTest`) for the shared form helpers and ensured the edit preview remains collapsed by default.
- Added **JVM mapper tests** (`ItemsMappersTest`) verifying `toMap`.
- Extended the **instrumentation tests** (`ItemFieldsComponentsTest`) to cover clearing the condition selection.

## Why
Reduce duplication between Add/Edit item flows, keep previews accurate, and let users reset the optional condition field without freely as was discussed in a previous PR.  
The mapper tests ensure serialization stays correct as the model evolves.

## Screenshots / Videos
New "clear" option for item condition :

https://github.com/user-attachments/assets/1d1a3c95-9f33-45d7-a025-cb40f04a6554

## Checklist
- [x] Tests: added or updated tests for `ItemFieldSharedComponents`, `ItemsMappersTest`, `EditItemsScreen`
- [x] Manual verification: app runs and basic item flows ran as normal

### CODE REVIEW

File: app/src/test/java/com/android/ootd/model/items/ItemsMappersTest.kt [hunk 1]
```diff
@@ -1,110 +1,84 @@
...
+        createItem(
+            material =
+                listOf(
+                    Material(name = "Cotton", percentage = 80.0),
+                    Material(name = "Polyester", percentage = 20.0)))
 
-    // Each required field missing -> null
-    keysToRemove.forEach { missing ->
-      val map = baseMap().toMutableMap()
-      map.remove(missing)
```
Comment (bbianca2004) [inline app/src/test/java/com/android/ootd/model/items/ItemsMappersTest.kt]: Quick question about some of the previous tests: I noticed that some of the previous tests in ItemMapperTest for negative edge cases were removed. Could I ask why you have decided to remove them? I am sure you know best or maybe they didn't well with the new implementation anymore, I was just wondering why.
